xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -224,6 +196,11 @@ public void updateDropPercentage(List<DropOverload> dropOverloads) {
       this.dropOverloads = checkNotNull(dropOverloads, "dropOverloads");
     }
 
+    @Override
+    public void setLoadStatsStore(@Nullable LoadStatsStore loadStatsStore) {
+      this.loadStatsStore = loadStatsStore;
```

Comment:
Ahh... Yeah, that's a problem. Updated the change.

Back to the problem: it is extremely complicated (realistically almost impossible) to cleanly handle load reporting for not losing a single load. The problem itself is illy-defined, it's never clear for how to deal with in-flight requests in the transition of toggling load reporting on/off.

To make it easier:  recording is always enabled, every single load will be recorded, even if load reporting is not enabled (as what we are used to do). Regarding the memory usage for recorded loads going unbounded will be a separate problem to consider (it's not easy for consequences of toggling load recording on/off).

Now the XdsClient is having two sets of APIs, one for load recording and one for load reporting:
```
LoadStatsStore addClientStats(String clusterName, @Nullable String clusterServiceName);
void removeClientStats(String clusterName, @Nullable String clusterServiceName)
```
is for creating the stats object used to record load. Put such an API on XdsClient instead of each EDS policy creating the stats object on its own aligns with the xDS LB policy refactoring design and make it work with our current routing hierarchy.

```
void reportClientStats();
void cancelClientStatsReport();
```
is for toggling load reporting on/off by LB policies. Yes, it. doesn't need any argument as you remember the loads for clusters being reported is driven by LRS responses and tracked stats: LRS server tells you which clusters to report loads for, LRS client looks up the currently tracked stats to find data for those clusters. Since stats data is tracked by `LoadStatsManager` now, the LRS client simply queries the `LoadStatsManager` to get stats.